### PR TITLE
fix: support try mode apply without prior config

### LINKED ISSUE #14044: apply config in try mode in maintenance would panic
Fix it!

### PR DESCRIPTION
**Pull request by smira**
Without this fix, Talos panics while trying to set the config to `nil` value. This is an edge case specifically for naving no machine config prior to the requested try operation.

Add a regression test in the maintenance test.

Fixes #14044

### CODE REVIEW

**Review by frezbo — APPROVED** [review]

**Review by Orzelius — APPROVED** [review]

File: internal/integration/provision/maintenance_basic.go [hunk 1]
```diff
@@ -110,6 +115,47 @@ func (suite *MaintenanceBasicSuite) TestAPI() {
...
+			Mode:           machine.ApplyConfigurationRequest_TRY,
+			TryModeTimeout: durationpb.New(tryModeTimeout),
+		})
+		suite.Require().NoError(err)
+
+		// now sysctl spec should be active
+		rtestutils.AssertResource(
+			suite.ctx, suite.T(), maintenanceClient.COSI, "proc.sys.fs.inotify.max_user_watches",
+			func(r *runtimeres.KernelParamSpec, assertion *assert.Assertions) {},
+		)
```
[the remark this answers is not in the document]
Comment (Orzelius) [reply internal/integration/provision/maintenance_basic.go:147]: yeah probably doesn't hurt to check the value as well
